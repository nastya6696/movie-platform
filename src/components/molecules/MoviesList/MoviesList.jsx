import {ErrorBoundary, MovieCard} from "../../atoms";

import styles from './styles.module.scss';

export const MoviesList = ({movies}) => {
  return (
    <div className={styles.moviesContainer}>
      {
        movies.map(movie =>
          <ErrorBoundary title={movie.title} key={movie.title}>
            <MovieCard movie={movie} />
          </ErrorBoundary>)
      }
    </div>
  )
};
