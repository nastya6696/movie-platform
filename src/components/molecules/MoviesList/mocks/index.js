export const Movies = [
  {

    year: 2004,
    type: 'Action & Adventure',
    imgName: 'PulpFiction'
  },
  {
    title: 'Bohemian Rhapsody',
    year: 2003,
    type: 'Drama, Biography, Music',
    imgName: 'BohemianRhapsody'
  },
  {
    title: 'Kill Bill: Vol 2',
    year: 1994,
    type: 'Oscar winning Movie',
    imgName: 'KillBill'
  },
  {
    title: 'Avengers: War of Infinity',
    year: 2004,
    type: 'Action & Adventure',
    imgName: 'Avengers'
  },
  {
    title: 'Inception',
    year: 2003,
    type: 'Action & Adventure',
    imgName: 'Inception'
  },
  {
    title: 'Reservoir dogs',
    year: 1994,
    type: 'Oscar winning Movie',
    imgName: 'DogsReservoir'
  }
]
