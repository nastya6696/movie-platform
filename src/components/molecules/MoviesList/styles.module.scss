@import "src/assets/scss/common";

.moviesContainer {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
}
