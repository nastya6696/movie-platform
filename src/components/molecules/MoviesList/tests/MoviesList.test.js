import {MoviesList} from "../MoviesList";
import {create} from 'react-test-renderer';


jest.mock('../../../atoms/ErrorBoundary', () => ({
  ErrorBoundary: (({children, title}) => `
     (--- ErrorBoundary ---): props {
      children: ${children},
      title: ${title}
     }`
  )
}));

describe('MoviesList', () => {
  it('should create simple MoviesList component', () => {
    const movies = [
      {
        id: 1,
        title: 'movie1'
      },
      {
        id: 2,
        title: 'movie2'
      },
      {
        id: 3,
        title: 'movie3'
      }
    ]
    const component = create(<MoviesList movies={movies}/>);

    expect(component).toMatchSnapshot();
  })
});
