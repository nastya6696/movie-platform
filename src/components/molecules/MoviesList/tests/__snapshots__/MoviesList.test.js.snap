// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList should create simple MoviesList component 1`] = `
<div
  className="moviesContainer"
>
  
     (--- ErrorBoundary ---): props {
      children: [object Object],
      title: movie1
     }
  
     (--- ErrorBoundary ---): props {
      children: [object Object],
      title: movie2
     }
  
     (--- ErrorBoundary ---): props {
      children: [object Object],
      title: movie3
     }
</div>
`;
